plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -636,12 +636,10 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                                 columns::get,
                                 columnId -> {
                                     IcebergColumnHandle column = columns.get(columnId);
-                                    Object prestoValue = deserializePartitionValue(
-                                            column.getType(),
-                                            partitionColumnValueStrings.get(columnId).orElse(null),
-                                            column.getName());
-
-                                    return NullableValue.of(column.getType(), prestoValue);
+                                    return partitionColumnValueStrings.get(columnId)
+                                            .map(value -> deserializePartitionValue(column.getType(), value, column.getName()))
```

Comment:
Can we add `@Nullable` to `deserializePartitionValue` method in a separated commit?